wled00/FX_fcn.cpp

```diff
@@ -860,6 +860,73 @@ void WS2812FX::fade_out(uint8_t rate) {
   }
 }
 
+/*
+* color adding function, adds colors while preserving the correct RGB ratio of the two combined
+*/
+uint32_t WS2812FX::AddPreserveRatio(uint32_t c1, uint32_t c2)
+  {
+    uint16_t r = R(c1) + R(c2);
+    uint16_t g = G(c1) + G(c2);
+    uint16_t b = B(c1) + B(c2);
+    uint16_t w = W(c1) + W(c2);
+    uint16_t max;
+
+   if (r>g && r>b && r>w) {
+      max = r;
+   } else if (g>b && g>w) {
+      max = g;
+   } else if (b>w){
+      max = b;
+   }else{
+      max = w;
+   }
+   if(max < 256){
+     return RGBW32(r, g, b, w);
+   }else{
+     float ratio = 255.0f / max;
+     return RGBW32(r * ratio, g * ratio, b * ratio, w * ratio);
+   }
+  }
+
+/*
+*  draw moving pixles with anti aliasing
+*  Inspired by Daves Garage Ep. 9 
+*  https://www.youtube.com/watch?v=RF7GekbNmjU&list=PLF2KJ6Gy3cZ7ynsp8s4tnqEFmY15CKhmH&index=7
+*/
+void WS2812FX::drawAntiAliasing(float fPos, float pixelSize, uint32_t color){
```

Comment:
I would recommend overloading setPixelColor() instead, using float for index and only do 1 pixel AA.

Reply:
haven't looked that much into the setPixelColor() function. If there's a way to not use floats without giving up functionality, then I'm open to ideas. The new effects DO depend on this function, but I'm sure they can be rewritten to use a different function for AA.